lib/port_scanner.py

```diff
@@ -0,0 +1,45 @@
+import queue
+import socket
+from threading import Thread, Lock
+from queue import Queue
+
+N_THREADS = 150
```

Comment:
I suggest using number of CPU threads instead of settings this constant (This could be done using the `nproc` comman on linux, or parsing the `/proc/cpuinfo`. Not sure on windows though). Also it would be cool to expose that as a option to the user

Reply:
You mean it should be possible to change how many threads there should be like we can do with ports?

Reply:
exactly. but the default should be the number of CPU threads available.

Reply:
Doing so many threads is actually harming the performance. You can try to decrease the threads to your cpu threads and it will get noticeably faster.

Reply:
okay

Reply:
what do you think about setting default with multiprocessing.cpu() count?

Reply:
And don't you think giving option to user how many threads they want to use a little stretch? Why would anyone need that much of control of a program that's few seconds utmost to execute?

Reply:
If `multiprocessing.cpu_count()` is available by default, let's use it. As for if it's a stretch to configure that, it's not. Most port scanners I've used had this as a option.

Reply:
okay, there's no need to install multiprocessing additionally btw
I'll handle that additional option rn

Reply:
## now it works like that. My default is 8 and I can choose as many as I want

![image](https://github.com/PixelToolkit/PixelToolkit/assets/113286903/0f7be706-cf9d-41e9-957f-62a1d359a60a)